tests/test_tree.py

```diff
@@ -53,6 +57,44 @@ def test_unserialized_tree_same_as_serialized(self):
         self.maxDiff = None
         self.assertEqual(before_load, after_load)
 
+        # leave the tree in a clean state
+        tree.clear()
+
+    def test_unserializable_tree(self):
+        """
+        Tests that if you store an unserializable object on an item, it will fail with a
+        "is not JSON serializable" TypeError.
+        """
+        # Indirectly create tasks, since we can't create them directly without a
+        # PublishPluginInstance object.
+        self.manager.collect_session()
```

Comment:
It seems to me that this test is doing so many things when a lot less could have been done.

Wouldn't something like:
```python
tree = self.manager.tree
item = tree.root_item.create_item("item.a", "Item A", "Item A")
item.properties["datetime"] = datetime.datetime.now()

from StringIO import StringIO
# Check that we get the error we expect.
with self.assertRaisesRegex(TypeError, "datetime.* is not JSON serializable"):
    tree.save(StringIO())
```

be enough as a test?